[nnc] Adding a test with dynamic shapes from a model

navahgar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72198

Differential Revision: [D33951741](https://our.internmc.facebook.com/intern/diff/D33951741)

navahgar committed:
[nnc] Adding a test with dynamic shapes from a model

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33951741).

eellison commented on test/cpp/tensorexpr/test_dynamic_shapes.cpp:
```diff
@@ -462,5 +462,85 @@ TEST(DynamicShapes, GraphWithCatAndBroadcast) {
+      -10, -9, -8, -7, -6, -5, -4, -3, -2};
+  TensorExprKernel k(graph, {}, symbolic_shape_inputs, false, symbolic_strides);
+
+  auto stmt = k.getCodeGenStmt();
```
unused

navahgar replied:
removed

eellison commented on test/cpp/tensorexpr/test_dynamic_shapes.cpp:
```diff
@@ -462,5 +462,85 @@ TEST(DynamicShapes, GraphWithCatAndBroadcast) {
+  stack.push_back(i4);
+  stack.push_back(i3);
+  stack.push_back(i2);
+  k.run(stack);
```
did we want to use the  other path  as well ? pre-allocated outputs?

navahgar replied:
I didn't add it earlier since I had that PR in a different stack. Now that has landed. So, added a check using that as well.

eellison reviewed (approved):
is there a reason to put this in cpp?

navahgar committed:
Update on "[nnc] Adding a test with dynamic shapes from a model"


Differential Revision: [D33951741](https://our.internmc.facebook.com/intern/diff/D33951741)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33951741).

navahgar committed:
Update on "[nnc] Adding a test with dynamic shapes from a model"


Differential Revision: [D33951741](https://our.internmc.facebook.com/intern/diff/D33951741)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33951741).

navahgar committed:
Update on "[nnc] Adding a test with dynamic shapes from a model"


Differential Revision: [D33951741](https://our.internmc.facebook.com/intern/diff/D33951741)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33951741).
